cmd/registry/controller/controller_test.go

```diff
@@ -225,37 +250,39 @@ func TestArtifacts(t *testing.T) {
 // Tests for aggregated artifacts at api level and specs as resources
 func TestAggregateArtifacts(t *testing.T) {
 	tests := []struct {
-		desc  string
-		setup func(context.Context, connection.Client, connection.AdminClient)
-		want  []*Action
+		desc string
+		seed []seeder.RegistryResource
+		want []*Action
 	}{
 		{
 			desc: "create artifacts",
-			setup: func(ctx context.Context, client connection.Client, adminClient connection.AdminClient) {
-				deleteProject(ctx, adminClient, t, "controller-test")
-				createProject(ctx, adminClient, t, "controller-test")
-				createApi(ctx, client, t, "projects/controller-test/locations/global", "test-api-1")
-				// Version 1.0.0
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1", "1.0.0")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1/versions/1.0.0", "openapi.yaml", gzipOpenAPIv3)
-				// Version 1.0.1
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1", "1.0.1")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1/versions/1.0.1", "openapi.yaml", gzipOpenAPIv3)
-				// Version 1.1.0
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1", "1.1.0")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-1/versions/1.1.0", "openapi.yaml", gzipOpenAPIv3)
-
-				// Test API 2
-				createApi(ctx, client, t, "projects/controller-test/locations/global", "test-api-2")
-				// Version 1.0.0
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2", "1.0.0")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2/versions/1.0.0", "openapi.yaml", gzipOpenAPIv3)
-				// Version 1.0.1
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2", "1.0.1")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2/versions/1.0.1", "openapi.yaml", gzipOpenAPIv3)
-				// Version 1.1.0
-				createVersion(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2", "1.1.0")
-				createSpec(ctx, client, t, "projects/controller-test/locations/global/apis/test-api-2/versions/1.1.0", "openapi.yaml", gzipOpenAPIv3)
+			seed: []seeder.RegistryResource{
+				// test api 1
+				&rpc.ApiSpec{
+					Name:     "projects/controller-test/locations/global/apis/test-api-1/versions/1.0.0/specs/openapi.yaml",
+					MimeType: gzipOpenAPIv3,
```

Comment:
Do we need all these MIME types specified, since we're creating specs that have empty contents? Aside from having a test that covers the filtering behavior I don't expect this field to be very useful in most tests.

Reply:
Thanks for pointing this out. I guess the MimeType usage got copied over as I used the same template from the initial controller tests. Removed the unnecessary usage.